File: sdks/java/lib/src/main/java/models/shared/MetadataTemplate.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * MetadataTemplate
 * A template for metadata that can be applied to files and folders
**/
public class MetadataTemplate {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("copyInstanceOnItemCopy")
    public Boolean copyInstanceOnItemCopy;
    public MetadataTemplate withCopyInstanceOnItemCopy(Boolean copyInstanceOnItemCopy) {
        this.copyInstanceOnItemCopy = copyInstanceOnItemCopy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("displayName")
    public String displayName;
    public MetadataTemplate withDisplayName(String displayName) {
        this.displayName = displayName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("fields")
    public MetadataTemplateMetadataFieldWrite[] fields;
    public MetadataTemplate withFields(MetadataTemplateMetadataFieldWrite[] fields) {
        this.fields = fields;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("hidden")
    public Boolean hidden;
    public MetadataTemplate withHidden(Boolean hidden) {
        this.hidden = hidden;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public MetadataTemplate withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("scope")
    public String scope;
    public MetadataTemplate withScope(String scope) {
        this.scope = scope;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("templateKey")
    public String templateKey;
    public MetadataTemplate withTemplateKey(String templateKey) {
        this.templateKey = templateKey;
        return this;
    }
    @JsonProperty("type")
    public MetadataTemplateTypeEnum type;
    public MetadataTemplate withType(MetadataTemplateTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/metadatatemplate.go
package shared

// MetadataTemplateMetadataFieldWriteMetadataOptionWrite
// An option for a Metadata Template Field.
//
// Options only need to be provided for fields of type `enum` and `multiSelect`.
// Options represent the value(s) a user can select for the field either through
// the UI or through the API.
type MetadataTemplateMetadataFieldWriteMetadataOptionWrite struct {
	ID  *string `json:"id,omitempty"`
	Key string  `json:"key"`
}

type MetadataTemplateMetadataFieldWriteTypeEnum string

const (
	MetadataTemplateMetadataFieldWriteTypeEnumString      MetadataTemplateMetadataFieldWriteTypeEnum = "string"
	MetadataTemplateMetadataFieldWriteTypeEnumFloat       MetadataTemplateMetadataFieldWriteTypeEnum = "float"
	MetadataTemplateMetadataFieldWriteTypeEnumDate        MetadataTemplateMetadataFieldWriteTypeEnum = "date"
	MetadataTemplateMetadataFieldWriteTypeEnumEnum        MetadataTemplateMetadataFieldWriteTypeEnum = "enum"
	MetadataTemplateMetadataFieldWriteTypeEnumMultiSelect MetadataTemplateMetadataFieldWriteTypeEnum = "multiSelect"
)

// MetadataTemplateMetadataFieldWrite
// A field within a metadata template. Fields can be a basic text, date, or
// number field, or a list of options.
type MetadataTemplateMetadataFieldWrite struct {
	Description *string                                                 `json:"description,omitempty"`
	DisplayName string                                                  `json:"displayName"`
	Hidden      *bool                                                   `json:"hidden,omitempty"`
	ID          *string                                                 `json:"id,omitempty"`
	Key         string                                                  `json:"key"`
	Options     []MetadataTemplateMetadataFieldWriteMetadataOptionWrite `json:"options,omitempty"`
	Type        MetadataTemplateMetadataFieldWriteTypeEnum              `json:"type"`
}

type MetadataTemplateTypeEnum string

const (
	MetadataTemplateTypeEnumMetadataTemplate MetadataTemplateTypeEnum = "metadata_template"
)

// MetadataTemplate
// A template for metadata that can be applied to files and folders
type MetadataTemplate struct {
	CopyInstanceOnItemCopy *bool                                `json:"copyInstanceOnItemCopy,omitempty"`
	DisplayName            *string                              `json:"displayName,omitempty"`
	Fields                 []MetadataTemplateMetadataFieldWrite `json:"fields,omitempty"`
	Hidden                 *bool                                `json:"hidden,omitempty"`
	ID                     *string                              `json:"id,omitempty"`
	Scope                  *string                              `json:"scope,omitempty"`
	TemplateKey            *string                              `json:"templateKey,omitempty"`
	Type                   MetadataTemplateTypeEnum             `json:"type"`
}
